src/st/class-wpml-pb-loader.php

```diff
@@ -12,6 +12,11 @@ public function __construct(
 
 		$page_builder_strategies = array();
 
+		$required = apply_filters( 'wpml_page_builder_support_required', array() );
```

Comment:
@strategio can you please add PHPDoc to the filter?

It would help to know where the corresponding `add_filter` is too: I can't find it anywhere in our code.

Reply:
Done @andreasciamanna! We use it in 2 places:
- `WPML_Page_Builders_Integration` which is used by Elementor and BB.
- `WPML_Gutenberg_Integration`

Reply:
@andreasciamanna, I know the link you shared. But what's really missing? I skipped the `@since` attribute because I have not accurrate idea when it was added (or I could put `Unknown`).

Should we really mention where `add_filter` is called? This is something that may change and the comment will not be updated.

Reply:
- The description `This filters the API page builder plugins that need to be supported` doesn't say much, does it?
I would write the actual reason for calling this filter.
- If the argument is not needed or ignored, that must be specified.
- `@since` can be skipped

> Should we really mention where add_filter is called? This is something that may change and the comment will not be updated.

The fact that it's called from two different places does not look right to me. I think it highlights a design flaw we should address in the future.  
I guess you can at least mention the two places where it's used as an example of implementation.

Reply:
@andreasciamanna, I improved the PHPDoc. Please check again!

I am not sure why we have 2 classes filtering this hook. My guess is that we started with 3rd part plugins which all have a common pattern (content store in metadata). When @brucepearson started to implement the support for Gutenberg, he found it in the mid way between shortcode and metadata page builders (Gutenberg store the content in the post body). So instead of changing the existing API `WPML_Page_Builders_Integration`, he decided to create a custom one for Gutenberg.